internal/user/user_profile.go

```diff
@@ -0,0 +1,31 @@
+package user
+
+import (
+	"strings"
+
+	"github.com/kopia/kopia/repo/manifest"
+)
+
+// Profile describes information about a single user.
+type Profile struct {
+	ManifestID manifest.ID `json:"-"`
+
+	Username     string `json:"username"`
+	PasswordHash string `json:"passwordHash"`
+}
+
+// SetPassword changes the password for a user profile.
+func (p *Profile) SetPassword(password string) error {
+	return p.setPasswordV1(password)
```

Comment:
Q: since the profile is encoded as JSON, does it make sense to explicitly encode the hashing scheme in a JSON field (for example, `"hashVersion" : "v1"`) vs. encoding it in the hash string itself? albeit slightly more verbose.

Reply:
done